applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py

```diff
@@ -0,0 +1,31 @@
+import KratosMultiphysics as Kratos
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+
+def CreateConvergenceCriteria(parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+    type = parameters["type"].GetString()
+    if type == "max_iter":
+        return MaxIterConvCriterium(parameters, optimization_problem)
+    else:
+        raise RuntimeError(f"CreateConvergenceCriteria: unsupported convergence type {type}.")
+
+class MaxIterConvCriterium(object):
+    @classmethod
+    def GetDefaultParameters(cls):
+        return Kratos.Parameters("""{
+            "type"              : "max_iter",
+            "max_iter"          : 0,
+        }""")
+
+    def __init__(self, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        self.__max_iter = parameters["max_iter"].GetInt()
+        self.__optimization_problem = optimization_problem
+
+    def CheckConvergence(self):
```

Comment:
```suggestion
    def IsConverged(self) -> bool:
```

I would rename it to `IsConverged` gives the purpose of the method. Ususally in checks, normally an error is thrown if check is failed.